arch/arm/core/aarch32/irq_manage.c

```diff
@@ -253,6 +259,34 @@ int irq_target_state_is_secure(unsigned int irq)
 	return NVIC_GetTargetState(irq) == 0;
 }
 
+/**
+ *
+ * @brief Disable and set all interrupt lines to target Non-Secure state.
+ *
+ * The function is used to set all HW NVIC interrupt lines to target the
+ * Non-Secure state. The function shall only be called fron Secure state.
+ *
+ * Notes:
+ * - All NVIC interrupts are disabled before being routed to Non-Secure.
+ * - Bits corresponding to un-implemented interrupts are RES0, so writes
+ *   will be ignored.
+ *
+ * @return N/A
+*/
+void irq_target_state_set_all_non_secure(void)
+{
+	for (uint8_t i = 0; i < sizeof(NVIC->ICER) / sizeof(NVIC->ICER[0]); i++) {
```

Comment:
Could you add a short comment before this and the next `for` loop? For those that don't know what ICER and ITNS do.

Reply:
done.